packages/terminal/src/node/terminal-backend-contribution.ts

```diff
@@ -39,6 +39,9 @@ export class TerminalBackendContribution implements MessagingService.Contributio
                 connection.onRequest('write', (data: string) => termProcess.write(data));
                 connection.onClose(() => output.dispose());
                 connection.listen();
+                termProcess.onClose(() => {
+                    connection.sendNotification('onClose', undefined);
```

Comment:
Why do we need it?

Reply:
this gives the front-end widget the opportunity to know the task has finished.